Mathlib/LinearAlgebra/CliffordAlgebra/SpinGroup.lean

```diff
@@ -0,0 +1,428 @@
+      simp only [HSMul.hSMul, SMul.smul, exists_exists_eq_and, exists_apply_eq_apply,
+        Submodule.mem_map, LinearMap.mem_range, DistribMulAction.toLinearMap_apply,
+        ConjAct.ofConjAct_toConjAct]
+    specialize hy1 hy2
+    have hx2 : ↑x * (↑y * (ι Q) b * ↑y⁻¹) * ↑x⁻¹ ∈ ConjAct.toConjAct x • LinearMap.range (ι Q) := by
+      simp only [HSMul.hSMul, SMul.smul, Units.mul_left_inj, Units.mul_right_inj,
+        exists_exists_eq_and, Submodule.mem_map, LinearMap.mem_range,
+        DistribMulAction.toLinearMap_apply, ConjAct.ofConjAct_toConjAct]
+      exact hy1
+    specialize hx1 hx2
+    rwa [hb] at hx1
+
+/-- This is another version of `lipschitzGroup.conjAct_smul_range_ι` which uses `involute`.-/
+theorem involute_act_ι_mem_range_ι [Invertible (2 : R)]
+    {x : (CliffordAlgebra Q)ˣ} (hx : x ∈ lipschitzGroup Q) (b : M) :
+      involute (Q := Q) ↑x * ι Q b * ↑x⁻¹ ∈ LinearMap.range (ι Q) := by
+  unfold lipschitzGroup at hx
+  induction hx using Subgroup.closure_induction'' generalizing b with
+  | mem x hx =>
+    obtain ⟨a, ha⟩ := hx
+    letI := x.invertible
+    letI : Invertible (ι Q a) := by rwa [ha]
+    letI : Invertible (Q a) := invertibleOfInvertibleι Q a
+    rw [LinearMap.mem_range, ← invOf_units x]
+    simp_rw [← ha, involute_ι]
+    refine ⟨-((⅟ (Q a) * QuadraticForm.polar Q a b) • a - b), ?_⟩
+    simp only [map_neg, neg_mul, ι_mul_ι_mul_invOf_ι Q a b]
+  | inv_mem x hx =>
+    obtain ⟨a, ha⟩ := hx
+    letI := x.invertible
+    letI : Invertible (ι Q a) := by rwa [ha]
+    letI : Invertible (Q a) := invertibleOfInvertibleι Q a
+    letI := invertibleNeg (ι Q a)
+    letI := Invertible.map (involute : CliffordAlgebra Q →ₐ[R] CliffordAlgebra Q) (ι Q a)
+    rw [LinearMap.mem_range, ← invOf_units x, inv_inv]
+    simp_rw [← ha, map_invOf, involute_ι, invOf_neg]
+    refine ⟨-((⅟ (Q a) * QuadraticForm.polar Q a b) • a - b), ?_⟩
+    simp only [map_neg, neg_mul, invOf_ι_mul_ι_mul_ι Q a b]
+  | one => simp only [Units.val_one, map_one, one_mul, inv_one, mul_one, LinearMap.mem_range,
+      exists_apply_eq_apply, forall_const]
+  | mul y z _ _ hy hz =>
+    simp only [Units.val_mul, map_mul, mul_inv_rev, LinearMap.mem_range]
+    let ⟨z', hz'⟩ := hz b
+    let ⟨y', hy'⟩ := hy z'
+    suffices
+        ∃ c : M, (ι Q) c = involute (Q := Q) ↑y * (involute (Q := Q) ↑z * (ι Q) b * ↑z⁻¹) * ↑y⁻¹ by
+      obtain ⟨p, hp⟩ := this
+      refine' ⟨p, by simp only [hp, mul_assoc]⟩
+    rw [← hz']
+    use y'
+
+
+theorem coe_mem_iff_mem {x : (CliffordAlgebra Q)ˣ} :
+    ↑x ∈ (lipschitzGroup Q).toSubmonoid.map (Units.coeHom <| CliffordAlgebra Q) ↔
+    x ∈ lipschitzGroup Q := by
+  simp only [Submonoid.mem_map, Subgroup.mem_toSubmonoid, Units.coeHom_apply, exists_prop]
+  norm_cast
+  exact exists_eq_right
+
+end lipschitzGroup
+
+/-- `pinGroup Q` is defined as the infimum of `lipschitzGroup Q` and `unitary (CliffordAlgebra Q)`.
+See `mem_iff`. -/
+def pinGroup (Q : QuadraticForm R M) : Submonoid (CliffordAlgebra Q) :=
```

Comment:
I wonder if this should be
```suggestion
def pinGroup (Q : QuadraticForm R M) : Subgroup (CliffordAlgebra Q)ˣ :=
```
(which requires quite a few changes later)

The benefit is that this then matches `lipschitzGroup`